src/modals/storage-connections/StorageConnectionsTable.js

```diff
@@ -0,0 +1,340 @@
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  wrappable,
+  EditableTextCell,
+  cancelCellEdits,
+  applyCellEdits,
+  validateCellEdits,
+  cellWidth,
+} from '@patternfly/react-table'
```

Comment:
In the new code the composable table is preferred - check https://www.patternfly.org/v4/components/table
The legacy table is model driven and adds unnecessary complexity.

@sjd78   since the code is ready I'm not sure if it's worth changing, what do you think?

Reply:
I have looked at Table Composable. It seems that Editable Rows are not supported there, and they are very much required here.

Reply:
The approach to editable rows is different, more React-compliant. Check i.e. my PR that uses them https://github.com/oVirt/ovirt-web-ui/pull/1549

Reply:
> In the new code the composable table is preferred - check https://www.patternfly.org/v4/components/table The legacy table is model driven and adds unnecessary complexity.
> 
> @sjd78 since the code is ready I'm not sure if it's worth changing, what do you think?

Good idea, but depends on if the composable table is available on the version of `@patternfly/react-core` we currently have in the project.

Reply:
> Check i.e. my PR that uses them https://github.com/oVirt/ovirt-web-ui/pull/1549

@rszwajko nice, but I find it very explicit. Is there a special reason that TableComposable doesn't have this implemented in a more generalized way, while the legacy Table does (with quirks though)?

> depends on if the composable table is available on the version of @patternfly/react-core we currently have in the project.

@sjd78 I think it is, but is it that bad to use legacy Table here? I would really prefer not to rewrite this whole thing :(

Reply:
@mkemel 
> nice, but I find it very explicit. Is there a special reason that TableComposable doesn't have this implemented in a more > generalized way, while the legacy Table does (with quirks though)?
> 
 
1. less code i.e. check `src/components/CreateVmWizard/steps/Networking.js` in the mentioned PR which is basically refactoring from legacy PF3 into composable PF4
2. no restrictions on widgets used in read/edit mode - legacy has basically only 2 hacked cell types (text and select) that need to fit into (informal) interface
3. no restrictions on validation and usage pattern
4. React-style prop-driven component without intermediate row model and all index-based logic

Reply:
@rszwajko Using the `TableComposable` would be nice.  @mkemel It is more explicit, but in a good way.  The model and cell renderers and cell editors are much more complex than just doing normal widgets in a table.  I just went through going form the legacy Table to `TableComposable` style and the edit logic stayed the same but the render complexity went down.

So @mkemel, do you have time to spend converting it over?  We will have both kinds of tables in the code base, but it would be preferred to have the current style in new code.  Your code would not need much rework.

Reply:
Having a working example now it might not be that horrible to rework that.

Reply:
Ack